khmer/_khmer.cc

```diff
@@ -176,7 +176,15 @@ static bool ht_convert_PyObject_to_HashIntoType(PyObject * value,
                             "k-mer length must equal the k-mer size");
             return false;
         }
-        hashval = ht->hash_dna(s.c_str());
+
+        try {
+            hashval = ht->hash_dna(s.c_str());
+        } catch (khmer_exception &e) {
+            PyErr_SetString(PyExc_ValueError, e.what());
+            Py_DECREF(val_as_str);
+            return false;
+        }
```

Comment:
I've passively seen code like this before, but I've never had a reason to pay close attention. With recent issues fresh on my mind, nice to see an example of C++ to Python exception handling.